src/providers.rs

```diff
@@ -165,31 +165,32 @@ pub fn do_update_provider(caller: Principal, update: UpdateProviderArgs) {
         let mut p = p.borrow_mut();
         match p.get(&update.provider_id) {
             Some(mut provider) => {
-                if provider.owner != caller && !is_authorized(&caller, Auth::Manage) {
+                if provider.owner != caller {
                     ic_cdk::trap("Provider owner != caller");
+                } else {
```

Comment:
(Wrapping in an `else` block to statically guarantee that this will only run for when authorized)